### ISSUE TITLE
Add support for vendor prefixed system fonts in `font-family-name-quotes`

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#1295]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Currently vendor prefixed "system fonts" such as `-apple-system` are not fully supported

> Which rule, if any, is this issue related to?

`font-family-name-quotes`

> What CSS is needed to reproduce this issue?

```css
a { font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif; }
```
** Depending on the `font-family-name-quotes` rule configuration the CSS font names above will be quoted differently, single, or double, or none etc

> What stylelint configuration is needed to reproduce this issue?

Both these configurations:
1. `"font-family-name-quotes": "single-unless-keyword"`
2. `"font-family-name-quotes": "double-unless-keyword"`

> Which version of stylelint are you using?

 `6.3.3`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI `stylelint test.css --config .stylelintrc`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No warnings to be flagged when `-apple-system` is not wrapped in quotes, single or double

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

Works as expected using:

• `"font-family-name-quotes": "single-where-required",`
• `"font-family-name-quotes": "single-where-recommended",`
• `"font-family-name-quotes": "double-where-recommended",`
• `"font-family-name-quotes": "double-where-required",`

Does *not* work as expected using:

1st case:
Rule: `"font-family-name-quotes": "single-unless-keyword"`
CSS:`a { font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif; }`
Result:
```shell
src/wp-includes/css/wp-embed-template.css
 1:15  ✖  Expected single quotes around font-family name "-apple-system"        font-family-name-quotes
 1:30  ✖  Expected single quotes around font-family name "BlinkMacSystemFont"   font-family-name-quotes
```

2nd case:
Rule: `"font-family-name-quotes": "double-unless-keyword"`
CSS:`a { font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif; }`
Result:
```shell
src/wp-includes/css/wp-embed-template.css
 1:15  ✖  Expected double quotes around font-family name "-apple-system"        font-family-name-quotes
 1:30  ✖  Expected double quotes around font-family name "BlinkMacSystemFont"   font-family-name-quotes
```

**Comment by ntwb** [comment]
_An aside fun fact and which is how the above started is if `-apple-system` is used as the first font in the stack when using the shorthand `font` property then the fonts won't be applied in Microsoft Edge_ 😖 

_The issue has been raised with Microsoft and we're waiting a response..._

**Comment by davidtheclark** [comment]
Is `-apple-system` the only possibility that we need to accommodate, or are there others?

**Comment by ntwb** [comment]
> Is `-apple-system` the only possibility that we need to accommodate, or are there others?

Via https://webkit.org/blog/3709/using-the-system-font-in-web-content/

```
font: -apple-system-body
font: -apple-system-headline
font: -apple-system-subheadline
font: -apple-system-caption1
font: -apple-system-caption2
font: -apple-system-footnote
font: -apple-system-short-body
font: -apple-system-short-headline
font: -apple-system-short-subheadline
font: -apple-system-short-caption1
font: -apple-system-short-footnote
font: -apple-system-tall-body
```

Consideration for `BlinkMacSystemFont` also may need to be taken into account.

Further reading:
• http://stackoverflow.com/a/32660790/1911294
• https://www.smashingmagazine.com/2015/11/using-system-ui-fonts-practical-guide/

**Comment by davidtheclark** [comment]
Ok, thanks @ntwb. Seems like we should detect fonts that start with `-[a-z]+-` and treat them the same way we treat keyword fonts like `serif` and `fantasy`. Does that sound right?

**Comment by jeddy3** [comment]
Sgtm

**Comment by ntwb** [comment]
> Seems like we should detect fonts that start with -[a-z]+- and treat them the same way we treat keyword fonts like serif and fantasy. Does that sound right?

Yes, sounds good 👍

**Comment by jeddy3** [comment]
Do we think this will come up again in another rule? If so, shall we pull this into a util i.e. `isSystemFontFamilyName()`?

**Comment by ntwb** [comment]
> An aside fun fact and which is how the above started is if -apple-system is used as the first font in the stack when using the shorthand font property which Apple requires then the fonts won't be applied in Microsoft Edge 😖

> The issue has been raised with Microsoft and we're waiting a response...

Just noting the response from Microsoft here:
> We have a fix for this internally and it is currently going through code review. It should roll out with Edge 14 in a couple months. There should be a preview build available in the insider track in a couple of weeks, too.

**Comment by ntwb** [comment]
> Do we think this will come up again in another rule? If so, shall we pull this into a util i.e. `isSystemFontFamilyName()`?

The quotes rule could use `isSystemFontFamilyName()` to ensure system fonts are not quoted?

**Comment by evilebottnawi** [comment]
@ntwb `isSystemFontFamilyName` return `true` or `false`, and in rule we return if `true`

**Comment by jeddy3** [comment]
Done in #1295
